source/common/router/config_impl.h

```diff
@@ -860,6 +872,21 @@ class RouteEntryImplBase : public RouteEntry,
     envoy::type::v3::FractionalPercent fractional_runtime_default_{};
   };
 
+  struct RouteStatsContext {
+    RouteStatsConfigSharedPtr route_stats_config_;
+    Stats::ScopeSharedPtr route_stats_scope_;
+    const Stats::StatNameManagedStorage& route_stat_name_storage_;
```

Comment:
we don't usually hold 'storage' pointers by ref. You might as well just hold the StatName here, as long as you are depending on another structure to own the storage.

Reply:
This is the structure that owns storage. There is no other structure.

Reply:
This is the structure that owns. I changed it from not using reference